client/component/people.js

```diff
@@ -54,49 +53,91 @@ function renderLayers(type, persona, profilePhotoLayers) {
     </View>
 }
 
-export function MyProfilePhoto({type='large', photo=null, faint=false, check=false, border=false}) {
+export function MyProfilePhoto({type='large', photo=null, faint=false, border=false}) {
     const personaPreview = usePersonaPreview();
     if (personaPreview?.photoUrl) {
         return <FaceImage photoUrl={photo ?? personaPreview.photoUrl} type={type} faint={faint} 
-            border={border} check={check} />
+            border={border} />
     } else if (personaPreview?.hue && personaPreview?.name) {
         return <LetterFace name={personaPreview.name} hue={personaPreview.hue} type={type} />
     } else {
         return <AnonymousFace faint={faint} type={type} border={border} />
     }
 }
 
-export function FaceImage({ face, photoUrl = null, type = 'small', faint = false, check = false, border = false}) {
-    
+export function UnverifiedBadge({size = 14}) {
+    const s = UnverifiedBadgeStyles;
+    return (
+        <View style={[s.container, { width: size, height: size, borderRadius: size / 2 }]}>
+            <Text style={[s.text, { fontSize: size === 24 ? 18 : 12 }]}>?</Text>
+        </View>
+    );
+}
+
+const UnverifiedBadgeStyles = StyleSheet.create({
+    container: {
+        backgroundColor: colorGreyHover,
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    text: {
+        color: 'black',
+    },
+});
+
+function Badge({type, name}) {
+    const datastore = useDatastore();
+    const fbUser = datastore.getFirebaseUser();
+    const personaPreview = usePersonaPreview();
+    const verificationStatus = personaPreview?.verificationStatus ?? null;
+    const badgeSize = badgeSizeMap[type ?? 'small'];
+
+    if (fbUser?.displayName !== name) {
+        return null;
+    }
+
+    if (verificationStatus === 'verified') {
```

Comment:
Note that the verification status is the status of their name, so we only want to show the verification badge if we are also showing their real name, rather than a pseudonym.

Right now the photo editing UI doesn't record whether the 'name' is a pseudonym. We could either have a field to check that, or we could look at the text and see if it contains any uppercase letters or spaces.

Reply:
i was solving for this case by checking (fbUser?.displayName !== persona?.name), assuming if the name is different from the firebase user name then they've set a psuedonym. does that not work?